.github/workflows/version.yml

```diff
@@ -0,0 +1,100 @@
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+          cache: npm
+      - run: npm ci
+      - name: Bump Version
+        id: inc-version
+        uses: actions/github-script@v6
+        env:
+          # list of labels that block bumping the version
+          STRICT_NO_RELEASE_TAGS: |
+            norelease
+          # list of labels that shouldn't bump the version by default
+          # adding a release label overrides this configuration
+          NO_RELEASE_TAGS: |
+            CI/CD
+            docs
+          BASE_VERSION: ${{ needs.master-version.outputs.version }}
+          PRERELEASE_TAG: beta
+        with:
+          result-encoding: string
+          script: |
+            const { execSync } = require('child_process');
+            const fs = require('fs');
+            const semver = require('semver');
+            const version = process.env.BASE_VERSION;
+            const { version: headVersion } = JSON.parse(fs.readFileSync('./package.json'));
+            const blockingTags = process.env.STRICT_NO_RELEASE_TAGS.split('\n');
+            const nonBlockingTags = process.env.NO_RELEASE_TAGS.split('\n');
+            const { labels } = context.payload.pull_request;
+
+            // get version level from PR label
+            const LABEL_PREFIX = 'release:';
+            const releaseLabels = labels
+              .map((label) => label.name)
+              .filter((label) => label.startsWith(LABEL_PREFIX));
+            if (releaseLabels.length > 1) {
+              throw new Error(`Found more than one release label: ${releaseLabels.join(', ')}`);
+            }
+            const releaseLevel = releaseLabels[0]?.replace(LABEL_PREFIX, '');
+            let shouldInc = !labels.some((label) => blockingTags.includes(label.name));
+            if (!releaseLevel && shouldInc) {
+              shouldInc = !labels.some((label) => nonBlockingTags.includes(label.name));
+            }
+            // increment version
+            const nextVersion = shouldInc ?
+              semver.inc(version, releaseLabels[0]?.replace(LABEL_PREFIX, '') || 'prerelease', process.env.PRERELEASE_TAG) :
+              version;
+
+            // write file
+            fs.writeFileSync('./package.json', JSON.stringify({ 
+              ...JSON.parse(fs.readFileSync('./package.json')), 
+              version: nextVersion
+            }, null, 2));
+            execSync('npx prettier --write package.json');
+
+            if (nextVersion !== headVersion) {
+              console.log(`Bumping version from ${version} to ${nextVersion}, current head version is ${headVersion}`);
+              return nextVersion;
+            } else {
+              console.log(`Version is up to date (${headVersion})`);
+              return '';
+            }
```

Comment:
in my opinion we should use the command  npm version patch/minor/major if we want to bump the version instead of writing our own script

Reply:
I wasn't aware that was an option. I will look into it

Reply:
But i m not sold on incrementing the version automatically yet. I think goin in the release section and push the button to release a new version is quick enough and still allow visibility on what are you publishin

Reply:
The question is what will happen if 2 PR incremented to the same tag, we merge one update from master...
Will it go mad?

Reply:
The impl I choose was to bump the version as part of the PR, avoiding more commits in the main history log

Reply:
I tried using `npm version`, it is the same as using semver with these exceptions:
- npm tags the commit
- npm needs the tree to be clean from changes, meaning that to use it in PRs we must revert the version of package.json, commit and then run `npm version`. That is an overkill.

Reply:
> But i m not sold on incrementing the version automatically yet. I think goin in the release section and push the button to release a new version is quick enough and still allow visibility on what are you publishing

The more CI the less headache of maintance and not only, it makes fabric more accessible to devs. Think of early adopters, think of an rc for each PR. In terms of getting devs to contribute that is a very good way IMO since a dev working on the rc can find their way around changes of a single PR and understand how to contribute a fix, and they will have the motivation to do so since they are rc devs. 
In general in will make fabric's development lifecycle a lot shorter and quicker.

Reply:
https://stackoverflow.com/questions/12278660/adding-tags-to-a-pull-request kills tagging in PRs for good

Reply:
It can be done in a separate commit after the PR is merged but I was under the impression we wanted to avoid that